org.jacoco.core.test/src-java8/org/jacoco/core/test/validation/BadCycleInterfaceTest.java

```diff
@@ -18,26 +18,21 @@
 /**
  * Test of "bad cycles" with interfaces.
  */
-public class BadCycleInterfaceTest extends BadCycleTestBase {
+public class BadCycleInterfaceTest extends ValidationTestBase {
 
 	public BadCycleInterfaceTest() throws Exception {
 		super("src-java8", BadCycleInterface.class);
 	}
 
 	@Test
 	public void test() throws Exception {
-		loader.loadClass(BadCycleInterface.Child.class.getName())
-				.getMethod("childStaticMethod").invoke(null);
-
-		analyze(BadCycleInterface.Child.class);
-
-		if (System.getProperty("java.version").startsWith("9-ea")) {
+		if (!System.getProperty("java.version").startsWith("9-ea")) {
 			// JDK-9042842
-			assertLine("2", ICounter.NOT_COVERED);
-		} else {
-			assertLine("2", ICounter.FULLY_COVERED);
+			assertLine("baseclinit", ICounter.FULLY_COVERED);
 		}
-		assertLine("4", ICounter.FULLY_COVERED);
+		assertLine("childclinit", ICounter.FULLY_COVERED);
+		assertLine("childdefault", ICounter.FULLY_COVERED);
+		assertLine("childstatic", ICounter.FULLY_COVERED);
```

Comment:
@marchof and I think that we should keep assertion for JDK 9 EA that lines are not executed - this way we can easily catch the moment when this will be fixed and will be able to remove conditional behavior:

```java
if (System.getProperty("java.version").startsWith("9-ea")) {
  // JDK-9042842
  assertLine("baseclinit", ICounter.NOT_COVERED);
  assertLine("childdefault", ICounter.NOT_COVERED);
} else {
  assertLine("baseclinit", ICounter.FULLY_COVERED);
  assertLine("childdefault", ICounter.FULLY_COVERED);
}
assertLine("childclinit", ICounter.FULLY_COVERED);
assertLine("childstatic", ICounter.FULLY_COVERED);
```